cmd/controller/sparsefilegenerator.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"github.com/golang/glog"
+	"github.com/openebs/node-disk-manager/pkg/util"
+
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+)
+
+/*
+Sparse File help simulate disk objects that can be used for testing and
+proto typing solutions built using node-disk-manager(NDM).  Sparse files
+will be created if NDM is provided with the location where sparse files
+should be located.
+
+On Startup, if a sparse directory (SPARSE_FILE_DIR) is specified as a
+environment variable, a Sparse file with specified size (SPARSE_FILE_SIZE) will
+be created and an associated Disk CR will be added to Kubernetes. By default
+only one sparse file will be created which can be changed by passing the desired
+number of sparse files required via the environment variable SPARSE_FILE_COUNT.
+
+On Shutdown, the status of the sparse file Disk CR will be marked as Unknown.
+*/
+
+const (
+	ENV_SPARSE_FILE_DIR   = "SPARSE_FILE_DIR"
+	ENV_SPARSE_FILE_SIZE  = "SPARSE_FILE_SIZE"
+	ENV_SPARSE_FILE_COUNT = "SPARSE_FILE_COUNT"
+
+	SPARSE_FILE_NAME          = "ndm-sparse.img"
+	SPARSE_FILE_DEFAULT_SIZE  = "1073741824"
+	SPARSE_FILE_DEFAULT_COUNT = "1"
+
+	SPARSE_DISKTYPE    = "sparse"
+	SPARSE_DISK_PREFIX = "sparse-"
+)
+
+// GetSparseFileDir returns the full path to the sparse
+//  file directory on the node.
+func GetSparseFileDir() string {
+
+	sparseFileDir := os.Getenv(ENV_SPARSE_FILE_DIR)
+
+	if len(sparseFileDir) < 1 {
+		return ""
```

Comment:
Will it be a good idea to have a default sparse dir as well (on similar lines as default count and default size) ? Also, create if not exists.

Reply:
Or is the idea to use sparse_file_dir as the decision flag to go ahead with sparse-file creation? If this is the idea, then its fine and you can ignore above comment.

Reply:
Yes, the idea is to use this also as a flag to create sparse files.